src/main/java/net/minestom/server/ServerFacade.java

```diff
@@ -0,0 +1,56 @@
+package net.minestom.server;
+
+import net.kyori.adventure.text.logger.slf4j.ComponentLogger;
+import net.minestom.server.advancements.AdvancementManagerProvider;
+import net.minestom.server.adventure.audience.AudienceManagerProvider;
+import net.minestom.server.adventure.bossbar.BossBarManagerProvider;
+import net.minestom.server.command.CommandManagerProvider;
+import net.minestom.server.event.GlobalEventHandlerProvider;
+import net.minestom.server.exception.ExceptionHandlerProvider;
+import net.minestom.server.extras.MojangAuthProvider;
+import net.minestom.server.gamedata.tags.TagManagerProvider;
+import net.minestom.server.instance.InstanceManagerProvider;
+import net.minestom.server.instance.block.BlockManagerProvider;
+import net.minestom.server.listener.manager.PacketListenerManagerProvider;
+import net.minestom.server.monitoring.BenchmarkManagerProvider;
+import net.minestom.server.network.ConnectionManagerProvider;
+import net.minestom.server.network.PacketProcessorProvider;
+import net.minestom.server.network.socket.ServerProvider;
+import net.minestom.server.recipe.RecipeManagerProvider;
+import net.minestom.server.scoreboard.TeamManagerProvider;
+import net.minestom.server.thread.ChunkDispatcherProvider;
+import net.minestom.server.timer.SchedulerManagerProvider;
+import net.minestom.server.world.DimensionTypeManagerProvider;
+import net.minestom.server.world.biomes.BiomeManagerProvider;
+
+public interface ServerFacade extends
```

Comment:
MinecraftServer, and I am extremely unconvinced about the need for the 60000000 other interfaces.

Reply:
By design, MinecraftServer should be optional to run Minestom. That is, nothing should depend on it, which is why these 60000000 interfaces are required. I would be happy to pass components directly without Providers, but I can’t due to many circular dependencies